src/views/Lending/Cooler/hooks/useConsolidateCooler.tsx

```diff
@@ -27,38 +27,96 @@ export const useConsolidateCooler = () => {
+
+    const tx = await v2Contract.setAuthorization(migratorAddress, authorizationDeadline);
+    const receipt = await tx.wait();
+    return receipt;
+  };
+
+  // Check if an address has authorized the migrator
+  const checkAuthorization = async (ownerAddress: string) => {
+    if (!provider) throw new Error("Please connect a wallet");
+
+    const migratorAddress = COOLER_V2_MIGRATOR_CONTRACT.getAddress(networks.MAINNET_HOLESKY);
+    const v2Contract = CoolerV2MonoCooler__factory.connect(
+      COOLER_V2_MONOCOOLER_CONTRACT.getAddress(networks.MAINNET_HOLESKY),
+      provider,
+    );
+
+    if (!migratorAddress) throw new Error("Missing migrator contract address");
+
+    const authorizationDeadline = await v2Contract.authorizations(ownerAddress, migratorAddress);
+    const currentTimestamp = Math.floor(Date.now() / 1000);
+
+    return authorizationDeadline.toNumber() > currentTimestamp;
+  };
+
   // Main mutation for consolidation
   const mutation = useMutation(
     async ({ coolers, newOwner }: { coolers: string[]; newOwner: string }) => {
       if (!signer) throw new Error("Please connect a wallet");
 
       const address = await signer.getAddress();
-
-      // 1. Get nonce from V2 contract for new owner
+      const migratorAddress = COOLER_V2_MIGRATOR_CONTRACT.getAddress(networks.MAINNET_HOLESKY);
       const v2Contract = CoolerV2MonoCooler__factory.connect(
         COOLER_V2_MONOCOOLER_CONTRACT.getAddress(networks.MAINNET_HOLESKY),
         provider,
       );
-      const nonce = await v2Contract.authorizationNonces(address);
-
-      // 2. Generate EIP-712 signature
-      const migratorAddress = COOLER_V2_MIGRATOR_CONTRACT.getAddress(networks.MAINNET_HOLESKY);
       const v2ContractAddress = v2Contract.address;
+
       if (!migratorAddress || !v2ContractAddress) throw new Error("Missing contract addresses");
 
-      //check if signer is the new owner
       const isNewOwner = address !== newOwner;
 
-      const { auth, signature } = await getAuthorizationSignature({
-        userAddress: address,
-        authorizedAddress: migratorAddress as `0x${string}`,
-        verifyingContract: v2ContractAddress as `0x${string}`,
-        chainId: networks.MAINNET_HOLESKY,
-        nonce: nonce.toString(),
-        signTypedDataAsync,
-      });
+      let auth;
+      let signature;
+
+      if (isNewOwner) {
+        // When newOwner is different, don't provide authorization
+        // The newOwner should call setAuthorization() separately
+        auth = {
+          account: "0x0000000000000000000000000000000000000000", // Zero address
+          authorized: migratorAddress,
+          authorizationDeadline: 0,
+          nonce: 0,
+          signatureDeadline: 0,
+        };
+        signature = {
+          v: 0,
+          r: "0x0000000000000000000000000000000000000000000000000000000000000000",
+          s: "0x0000000000000000000000000000000000000000000000000000000000000000",
+        };
+      } else {
```

Comment:
this is necessary since setAuthorization is being called from the new EOA outside of this tx and we won't have that signature.